sdk/core/core/test/cmocka/test_json_builder.c

```diff
@@ -150,31 +150,88 @@ void test_json_builder(void** state)
             "\"array\":[1,2,\"sd\":{},3]"
             "}")));
   }
-
-  (void)state;
```

Comment:
Can you explain what this was supposed to be for and why we can now remove it?

Reply:
sure, I removed it from here because it was already above on line 20.
Basically, each test we create for cmocka need to follow this signature
```c
void name_of_test(void ** state) {
}
```
and since we are not using state, we need to void it. Otherwise compiler will fail and tell us that state is not used at all.

you might wonder what this state is used for:
https://api.cmocka.org/group__cmocka__exec.html#ga4d54508b0b7150afb27b4da38f0015c6

We can basically create some state (structure) and all tests will get a pointer to the pointer of that struct (state)